core/sr-primitives/src/traits.rs

```diff
@@ -333,6 +333,24 @@ impl CheckEqual for substrate_primitives::H256 {
 	}
 }
 
+impl<I> CheckEqual for I where I: DigestItem {
+	#[cfg(feature = "std")]
+	fn check_equal(&self, other: &Self) {
+		if self != other {
+			println!("DigestItem: given={:?}, expected={:?}", self, other);
```

Comment:
Wouldn't that code also land in production code? Shouldn't we just implement `PartialEq` and do the printing just in our test code?

Reply:
It will. The same is true for the storage root - is is checked && printed in production code as well. Afaiu this would help in diagnosing issues in production environment.

`DigestItem` is already `PartialEq`